### PR TITLE
docs: Update documentation with final module shape and usage

### PR DESCRIPTION
**Pull request by paolobueno**
## Motivation
Update core.adoc with current sdk structure

### CODE REVIEW

File: packages/core/README.md [hunk 1]
```diff
@@ -1,4 +0,0 @@
-  TODO DOCS
```
Comment (paolobueno) [inline packages/core/README.md]: I realized this would look very similar to a converted version of core.adoc.

Personally I'd like to avoid duplicating content, but the package's readme is going to be visible in NPM. Should it maybe contain a link to the hosted documentation? Not sure.

File: README.md [hunk 2]
```diff
@@ -15,8 +23,7 @@ AeroGear SDK for Java Script, Apache Cordova and React Native.
 
 AeroGear Services SDK consist of set of separate SDKs
 
-- [Core](./docs/core/README.adoc): Common base for all SDKs
-- [Metrics](./docs/metrics/README.adoc): Mobile application metrics SDK
+- [Core](./docs/core.adoc): Common base for all SDKs
```
Comment (josemigallas) [inline README.md]: @wtrocki @paolobueno are we going to have metrics separated in the end? Let's add `Auth` too, no?

File: docs/core.adoc [hunk 3]
```diff
@@ -10,22 +10,60 @@ It's main features are:
 
 == Installation
 
-Use the Cordova CLI to add the Core module as a plugin to your project:
+The `@aerogear/core` module is available via NPM:
 
 ----
-cordova plugin add @aerogearservices/core-cordova
+npm install @aerogear/core
```
Comment (josemigallas) [inline docs/core.adoc]: `--save` :P

File: docs/core.adoc [hunk 4]
```diff
@@ -10,22 +10,60 @@ It's main features are:
...
+----
+
+This will install the packages with these native dependencies from the NPM registry and properly link all native dependencies to the project's cordova Platforms.
 
 Refer to the link:./getting-started[Getting Started] guide for more information on installing the AeroGear SDK.
 
 == Usage
 
-After installation, the Core module will become available through the global `window.aerogear` object in your project.
+The Core module supports parsing of the configuration file generated by the mobile CLI:
```
Comment (josemigallas) [inline docs/core.adoc]: This has changed a little since yesterday, now the parser will be transparent for the user. It will only be required to import the JSON file and pass it to the constructor:
```js
import { MyService } from "@aerogear/my-module";
import appConfig from "../mobile-services.json";

const service = new MyService(appConfig);
```

File: docs/core.adoc [hunk 5]
```diff
@@ -10,22 +10,60 @@ It's main features are:
...
+import configuration from '../mobile-services.json';
+const metrics = new MetricsService(configuration);
+metrics.sendAppAndDeviceMetrics().then(() => {
+  console.info("Metrics registered successfully");
+});
+----
+
+App metrics record:
+
+- *ID*: The identifier of your App
```
Comment (josemigallas) [inline docs/core.adoc]: I think this one is `clientId`.

File: packages/core/README.md (hunk 1 above)
Comment (josemigallas) [reply packages/core/README.md]: Yeah it would be more user friendly to have the info right on NPM, we could simply put some usage instructions and API reference in the Readme, maybe a generated js-doc.

File: docs/core.adoc (hunk 3 above)
Comment (wtrocki) [reply docs/core.adoc]: Depending on the version of NPM save will be default option.

File: README.md (hunk 2 above)
Comment (wtrocki) [reply README.md]: Metrics is separate docs.

File: packages/core/README.md (hunk 1 above)
Comment (wtrocki) [reply packages/core/README.md]: Let's leave that for now. Ideally readme should just point to documentation to not replicate content. I have the same issue on IOS. Cocoapods

File: README.md (hunk 2 above)
Comment (ciaranRoche) [reply README.md]: @wtrocki We have a PR here covering auth documentation https://github.com/aerogear/aerogear-js-sdk/pull/39

Comment (wtrocki) [reply README.md]: Great. Let's split metrics then.

File: docs/core.adoc (hunk 4 above)
Comment (wtrocki) [reply docs/core.adoc]: @josemigallas Let's propose some update here to simplify @paolobueno work

File: docs/core.adoc (hunk 5 above)
Comment (wtrocki) [inline docs/core.adoc]: Let's not include that. This is internal implementation detail.

File: docs/core.adoc (hunk 3 above)
Comment (josemigallas) [reply docs/core.adoc]: That's an even better reason to add it.

Comment (paolobueno) [reply docs/core.adoc]: `--save` has been on by default since npm 5, and latest nodejs LTS (carbon) uses 5.6: https://nodejs.org/en/download/releases/.

Current `npm help install` doesn't even list `--save` as a valid flag anymore, just the different flavors (`--save-dev, --no-save, etc`) so my vote is to keep it out of new documentation.

File: docs/core.adoc (hunk 4 above)
Comment (paolobueno) [reply docs/core.adoc]: :+1: I guess there's really not much 'usage' for the core module, I'll reword this as I split away the metrics part

File: docs/core.adoc [hunk 6]
```diff
@@ -10,22 +10,34 @@ It's main features are:
...
 
-=== Networking
+[source, javascript]
+----
+import { MetricsService } from '@aerogear/core';
+import configuration from '../mobile-services.json';
 
-=== Metrics Service
+// utilizes Core to retrieve it's own configuration
+const metrics = new MetricsService(configuration);
```
Comment (paolobueno) [inline docs/core.adoc]: @josemigallas @wtrocki  I've tried to change the Usage section to point out how core's used by other modules and doesn't make sense by itself but it can be a bit clunky, removing this section is also an option.
